### PR TITLE
Fix #791 body_reader.js#createBodyWriter() may corrupt on multi-bytes character

### LINKED ISSUE #791: body_reader.js#createBodyWriter() may corrupt on multi-bytes character
Multi-bytes character like Chinese use two or three or four bytes to encode. When we post request with large body, underly TCP may transfer it over multiple packets.

When TCP split packet in the middle of a multi-bytes character, following code may broken and corrupt the output:
```javascript
//@@ -27,9 +27,9 @@ function createBodyWriter(req) {
30        req.body = '';
31        return (function (chunk) {
32            req.body += chunk.toString('utf8');
33        });
```

Fix:
Use `Buffer` and concat them, then on request done convert at once may fix this issue.

### PR DESCRIPTION
**Pull request by soplwang**
Use Buffer and concat them, then on request done convert at once to fix
this issue.

### CODE REVIEW

File: lib/plugins/body_reader.js [hunk 1]
```diff
@@ -75,6 +75,9 @@ function bodyReader(options) {
                 return;
             }
 
+            if (bodyWriter.name === 'ub') {
```
Comment (yunong) [inline lib/plugins/body_reader.js]: Why are you checking for the name of the bodywriter here?

File: test/plugins.test.js [hunk 2]
```diff
@@ -400,6 +400,33 @@ test('body json ok (no params)', function (t) {
     });
 });
 
+test('body json ok (large)', function (t) {
```
Comment (yunong) [inline test/plugins.test.js]: Could you prefix this test name with GH-792?

Comment (soplwang) [inline test/plugins.test.js]: OK

File: lib/plugins/body_reader.js (hunk 1 above)
Comment (soplwang) [inline lib/plugins/body_reader.js]: I want to minimize the changes and, align the text chunk handling with the other hand binary body's. I choose use bodywriter's name to distinct these two mode buffers.

At here, with check of bodywriter's name, I can know the buffer type. If buffer is actual utf8 mode text, then convert it to 'String' to compatible with intent behavior.

**Comment by soplwang** [comment]
dup with #792, but has minimal changes

**Comment by yunong** [comment]
Closing as dupe of #775

**Pull request closed without merging**